StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1504AllAccessorsMustBeSingleLineOrMultiLine.cs

```diff
@@ -72,18 +76,50 @@ public class SA1504AllAccessorsMustBeSingleLineOrMultiLine : DiagnosticAnalyzer
             ImmutableArray.Create(Descriptor);
 
         /// <inheritdoc/>
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(this.HandlePropertyDeclaration, SyntaxKind.PropertyDeclaration, SyntaxKind.IndexerDeclaration, SyntaxKind.EventDeclaration);
+        }
+
+        private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
         {
-            get
+            var property = (BasePropertyDeclarationSyntax) context.Node;
+
+            if (property.AccessorList == null 
+                || property.AccessorList.IsMissing
+                || property.AccessorList.Accessors.Count != 2)
+            {
+                return;
+            }
+
+            if (!this.AreAllAccessorsInMultiLines(property.AccessorList.Accessors) 
+               && !this.AreAllAccessorsInOneLine(property.AccessorList.Accessors))
             {
-                return SupportedDiagnosticsValue;
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, property.AccessorList.Accessors[1].GetLocation()));
             }
+
         }
 
-        /// <inheritdoc/>
-        public override void Initialize(AnalysisContext context)
+        private bool AreAllAccessorsInOneLine(IEnumerable<AccessorDeclarationSyntax> accessors)
```

Comment:
:bulb: Consider creating a single method:

```csharp
private static bool IsSingleLineAccessor(AccessorDeclarationSyntax syntax);
```

You can then use `x.All(IsSingleLineAccessor)` and `!x.Any(IsSingleLineAccessor)` instead of two methods that duplicate the logic.